WMIAdventure/backend/WMIAdventure_backend/battle/migrations/insert_test_players_data.py

```diff
@@ -0,0 +1,92 @@
+# Generated by Django 3.2.1 on 2021-06-07 20:48
+import random
+
+from django.db import migrations
+
+from IngameUsers.models import UserProfile, Semester, Deck
+from cards.models import Card, CardInfo, CardLevel, CardEffect, CardLevelEffects
+from users.models import User
+
+
+def create_test_cards():
+    """
+    Creates test cards if there are less than 5 cards in the database.
+    At least 5 cards are required to create test users.
+
+    :return: None
+    """
+    cards_count = len(Card.objects.all())
+    while cards_count < 5:
+        create_test_card(cards_count)
+        cards_count += 1
+
+
+def create_test_card(cards_count: int):
+    """
+    Creates test card.
+
+    :param cards_count: How many cards are in the database. Will be used to generate unique test card name.
+    :return: None
+    """
+    test_card_info = CardInfo.objects.create(name=f"test_card{cards_count}", tooltip="test_card")
+    test_card = Card.objects.create(info=test_card_info, level=CardLevel.objects.get(pk=1))
+    test_card.effects.create(card_effect=CardEffect.objects.get(pk=CardEffect.EffectId.DMG),
+                             target=CardLevelEffects.Target.OPPONENT, power=cards_count, range=cards_count)
+
+
+def create_test_players_data(apps, schema_editor):
+    """
+    Creates data in database for test users. These test users will be used to test battle API calls.
+
+    :param apps:
+    :param schema_editor:
+    :return: None
+    """
+
+    create_test_cards()
+
+    # Create 5 test players.
+    test_users_count = 5
+    for test_player_num in range(test_users_count):
+        create_test_player_data(test_player_num)
+
+
+def create_test_player_data(player_num: int):
+    """
+    Creates all data in database for test users to be able to fight in battle.
+    Creates test user's deck and all data needed to create deck.
+
+    :param player_num: Test player num, must be unique, will be used to create unique name, email, etc.
+    :return: None
+    """
+    # Create test User
+    test_user = User.objects.create_user(f"test_user{player_num}", f"test_user{player_num}@company.com")
```

Comment:
Zanim stworzę usera "test_user1" powinienem sprawdzać, czy taki już istnieje? Jeśli już taki istnieje, to powinienem go usuwać?

Reply:
Tak samo jak wyżej, dobrze byłoby jakby Ci userzy mieli jakieś wiarygodne nazwy

Reply:
Spoko, zrobię listę jakichś fikcyjnych usernamów